### PR TITLE
Batch uploader

### PR DESCRIPTION
**Pull request by oliaboo**
upload_datapoints changed to upload in batches
input for upload_datapoints changed from generator to list
tests adjusted

### CODE REVIEW

File: parsers/uploader.py [hunk 1]
```diff
@@ -35,15 +36,23 @@ def post(data, endpoint=UPLOAD_URL, token=UPLOAD_API_TOKEN):
                          headers={'API_TOKEN': token})
 
 
-def upload_datapoints(gen, upload_func=post):
-    """Save data from generator *gen* to database endpoint.
+def upload_datapoints(ust_data, upload_func=post):
```
Comment (epogrebnyak) [inline parsers/uploader.py]: Must keep generic name - this fucntion is not just for `ust_data`

File: parsers/uploader.py [hunk 2]
```diff
@@ -35,15 +36,23 @@ def post(data, endpoint=UPLOAD_URL, token=UPLOAD_API_TOKEN):
...
        Returns:
           True on success (status code 200),
           False otherwise
     """
-    response = upload_func(data=to_json(gen))
-    if response.status_code == 200:
-        return True
-    else:
-        return False
+    ust_data_chunks = [ust_data[i:i + 1000] for i in range(0, len(ust_data), 1000)]
```
Comment (epogrebnyak) [inline parsers/uploader.py]: I think 1000 should be a constant like chunck_size = 1000

File: parsers/uploader.py [hunk 3]
```diff
@@ -35,15 +36,23 @@ def post(data, endpoint=UPLOAD_URL, token=UPLOAD_API_TOKEN):
...
+    for ust in ust_data_chunks:
+        json_ust_data = to_json(ust)
+        num_of_attempts = 0
+        while(num_of_attempts < 5):
```
Comment (epogrebnyak) [inline parsers/uploader.py]: same here - `max_attempts` should be a variable, `delay = 10` too

File: parsers/runner.py [hunk 4]
```diff
@@ -122,7 +124,7 @@ class USTbonds(ParserBase):
     source_url = "https://www.treasury.gov/resource-center/data-chart-center/interest-rates/Pages/TextView.aspx?data=yield"
 
     def all_items(self):
-        return ust.yield_ust_dict(self.start)
+        return ust.get_ust_dict(self.start)
```
Comment (epogrebnyak) [inline parsers/runner.py:127]: Switching from iterator to list makes this a little different from other parsers. Can keep the list, but it makes parsers a little different one from another.

**Review by epogrebnyak — APPROVED** [review]

File: parsers/uploader.py (hunk 2 above)
Comment (oliaboo) [reply parsers/uploader.py]: constants added

File: parsers/uploader.py (hunk 3 above)
Comment (oliaboo) [reply parsers/uploader.py]: added

File: parsers/runner.py (hunk 4 above)
Comment (oliaboo) [reply parsers/runner.py:127]: not sure for now

**Pull request merged**